### DISCUSSION TITLE
Relationship between inline post and diag_table?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [ufs-community/ufs-weather-model#1622] [Q&A] [answered]
Does turning off variables in diag_table impact what is produced by inline post? I've been winnowing down what is written out via diag_table and in my last test run no grib output files were produced. I'm wondering if I turned them off somehow editing diag_table or if something else happened. Thanks!

**Comment by pjpegion** [comment, marked as answer]
The diag_table dictates what is passed to inline post, so removing fields from there will affect inline post. I have found that if you remove too many fields, inline post dies. I forget which field it really needs, I think it is the pressure thickness.

**Reply by benjamin-cash** [reply]
@pjpegion do you happen to know if this is documented somewhere? I don't really want to write out a ton of fields I don't need but also don't want to do detection by casualty to find that critical field either. ;)

**Reply by benjamin-cash** [reply]
@gspetro-NOAA - Here is the question I am looking for help with. I removed a number of variables from diag_table, and inline post has stopped producing output entirely. I need someone to help me understand the link between diag_table and inline post. :)

**Reply by gspetro-NOAA** [reply]
@benjamin-cash I have reached out to @fossell and @hertneky to see if they can help with this. In the meantime, there are the UPP [Doxygen files](https://noaa-emc.github.io/UPP/), which provide more detail than the general User's Guide, and you could try to parse through the [post_fv3.F90 file](https://github.com/NOAA-EMC/fv3atm/blob/d9e619ed4a5e685b3558e0b1bf6929880960864b/io/post_fv3.F90), since that seems to be throwing the errors (based on your comment below). That said, I'm hope Kate and/or Tracy may be able to come to the rescue with a clearer answer! :)

**Reply by benjamin-cash** [reply]
Thanks @gspetro-NOAA! Going into the code and looking at the error message seems to have done the trick - it turns out that if you comment out the land fraction in diag_table it kills inline post completely. Not exactly intuitive, but all's well that ends well. :)

**Reply by fossell** [reply]
@benjamin-cash - Just to follow up and reiterate what @pjpegion mentioned, the diag_table lists the fields that are _input_ into the inline post library from the model, it is not a list of output variables from inline post library (UPP). Removing entries from the diag_table will not remove them from your output files, and could cause UPP to crash because it needs certain fields from the model to run and compute diagnostics. The subset of minimum required variables is not presently documented, so best to assume UPP needs them all. The postxconfig-NT*.txt files and corresponding xml files list output fields for various applications. [This section](https://upp.readthedocs.io/en/latest/InputsOutputs.html#inputs-and-outputs) of the Users Guide has some (though probably not all) information about the inputs and outputs and how to control output fields for offline post which might be helpful.

**Reply by benjamin-cash** [reply]
Hi @fossell - In this case I wasn't trying to remove files being produced by UPP, but rather cut down on the number of variables being written in the atmf* files. Since I wasn't planning to retain most of those variables and storage volume is an issue I was hoping to stop writing most of them - removing them from diag_table _does_ remove them from the atmf* files. Once I looked at the code where the error message was coming from I was able to determine what the critical missing field was and restart.

**Reply by fossell** [reply]
Ah, ok, I understand now. Glad you found the problematic field and still able to reduce file size.

**Comment by pjpegion** [comment]
Not that I am aware of

On Thu, Feb 23, 2023 at 11:13 AM benjamin-cash ***@***.***>
wrote:

> @pjpegion <https://github.com/pjpegion> do you happen to know if this is
> documented somewhere? I don't really want to write out a ton of fields I
> don't need but also don't want to do detection by casualty to find that
> critical field either. ;)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/discussions/1622#discussioncomment-5091705>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJIRVJFHIPMVXJPYPFCPZX3WY6SF3ANCNFSM6AAAAAAVF6ECNM>
> .
> You are receiving this because you were mentioned.Message ID:
> <ufs-community/ufs-weather-model/repo-discussions/1622/comments/5091705@
> github.com>
>


-- 
Phil Pegion (he/him/his)
Physical Scientist
NOAA/Physical Sciences Laboratory
(303) 497-7897
***@***.***

**Comment by pjpegion** [comment]
Do you see any error in standard out like: 'fail to read'
I think the problem is upp is having an error then hits a FORTRAN STOP, but this only stops that processor, and does not to an mpi_abort, so the error will way up in the output file, in the area where the 1st history files are written.

**Reply by benjamin-cash** [reply]
There are a bunch of error messages along this lines of: ERROR            PET1245 post_fv3.F90:917  - Passing error in return code

but nothing I can see that tells me what specific field was missing. I haven't gone into the code yet to see if that line gives me the answer.

**Reply by pjpegion** [reply]
that is associated with post not finding ice fraction or land fraction

**Reply by benjamin-cash** [reply]
Yep, and I had turned off land fraction, because why do I need to write land fraction every 6h? I've turned it back on to see if that at least gets me to a different error.
